### PR TITLE
Rnn bias deprecation

### PR DESCRIPTION
**Pull request by lungd**
This PR deprecates RNNCell.bias to make it consistent with base.jl
In other ML frameworks it is possible to deactivate the bias of other recurrent layers as well (e.g. LSTM).
I could add a `use_bias` without `bias` to LSTM and GRU but I think in this case a deprecated `bias` would be a better choice so you get the same behavior for all the layers (i.e. you can create RNNCells and LSTMs with `bias`, which prints a warning)

### CODE REVIEW

**Comment by avik-pal** [comment]
> I could add a use_bias without bias to LSTM and GRU but I think in this case a deprecated bias would be a better choice so you get the same behavior for all the layers (i.e. you can create RNNCells and LSTMs with bias, which prints a warning)

Let's do that in a separate PR. Also go with `use_bias` since users should not be using bias kwarg

**Comment by lungd** [comment]
Waiting for approval

File: src/layers/recurrent.jl
Comment (avik-pal) [inline src/layers/recurrent.jl:52]: Can you add a msg like https://lux.csail.mit.edu/dev/devdocs/style_guide/#deprecation so that we don't miss out things while deleting things?

File: src/layers/recurrent.jl
Comment (avik-pal) [inline src/layers/recurrent.jl:123]: I think keeping this as `bias` is better. But if we are making this change, we need to do it for all the bias optional layers.

**Review by avik-pal — CHANGES_REQUESTED** [review]
Also, version bump.

File: src/layers/recurrent.jl
Comment (lungd) [reply src/layers/recurrent.jl]: This was actually an unintended change, I've fixed now

File: src/layers/recurrent.jl
Comment (avik-pal) [reply src/layers/recurrent.jl:52]: Add this comment `# Deprecated Functionality (Remove in v0.5)`

Comment (lungd) [reply src/layers/recurrent.jl:52]: The constructor does already warn about the deprecated kwarg similar to Dense. The testset contains a nested 'Deprecations' testset. I've now added a message about the new `use_bias` kwarg to the changelog.

Do you want me to move the `Base.depwarn` from src/layers/recurrent.jl to src/deprecated.jl?

Comment (avik-pal) [reply src/layers/recurrent.jl:52]: No everything else is fine. Just add the comment above the `if !dismissing(bias)` like.

Comment (lungd) [reply src/layers/recurrent.jl:53]: done

**Review by avik-pal — APPROVED** [review]

**Pull request merged**
